src/main.ts

```diff
@@ -65,15 +57,26 @@ export async function main(){
         - voulme mount .azure session token file between host and container,
         - volume mount temp directory between host and container, inline script file is created in temp directory
         */
-        let command: string = `run --workdir ${process.env.GITHUB_WORKSPACE} -v ${process.env.GITHUB_WORKSPACE}:${process.env.GITHUB_WORKSPACE} `;
-        command += ` -v ${process.env.HOME}/.azure:/root/.azure -v ${TEMP_DIRECTORY}:${TEMP_DIRECTORY} `;
-        command += ` ${environmentVariables} `;
-        command += `--name ${CONTAINER_NAME} `;
-        command += ` mcr.microsoft.com/azure-cli:${azcliversion} ${startCommand}`;
+        let command: string = `run`;
+        var args = ["--workdir", `${process.env.GITHUB_WORKSPACE}`,
+            "-v", `${process.env.GITHUB_WORKSPACE}:${process.env.GITHUB_WORKSPACE}`,
+            "-v", `${process.env.HOME}/.azure:/root/.azure`,
+            "-v", `${TEMP_DIRECTORY}:${TEMP_DIRECTORY}`
+        ];
+        for (let key in process.env) {
+            if (!checkIfEnvironmentVariableIsOmitted(key) && process.env[key]) {
+                args = args.concat("-e", `${key}=${process.env[key]}`);

```

Comment:
[`concat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) creates a new array every time. We may consider using [`push`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push).